core/app/models/spree/customer_return.rb

```diff
@@ -61,7 +61,7 @@ def generate_number
     end
 
     def return_items_belong_to_same_order
-      if return_items.select{ |return_item| return_item.inventory_unit.order_id != order_id }.any?
+      if return_items.select{ |return_item| return_item.inventory_unit.order != order }.any?
```

Comment:
This does not work, as the object being validated might have been changed in-memory and the DB query won't catch those changes.